File: auth/src/main/java/com/diagorn/sparkathon/auth/config/KafkaTopicConfig.java
package com.diagorn.sparkathon.auth.config;

import com.diagorn.sparkathon.auth.config.properties.KafkaTopicProperties;
import lombok.RequiredArgsConstructor;
import org.apache.kafka.clients.admin.NewTopic;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.config.TopicBuilder;

@Configuration
@RequiredArgsConstructor
@EnableConfigurationProperties(KafkaTopicProperties.class)
public class KafkaTopicConfig {

    private final KafkaTopicProperties kafkaTopicProperties;

    @Bean
    public NewTopic newUserTopic() {
        return TopicBuilder.name(kafkaTopicProperties.getNewUser()).build();
    }

    @Bean
    public NewTopic editUserTopic() {
        return TopicBuilder.name(kafkaTopicProperties.getEditUser()).build();
    }
}

File: auth/src/main/java/com/diagorn/sparkathon/auth/config/KafkaTopicConfig.kt
package com.diagorn.sparkathon.auth.config

import com.diagorn.sparkathon.auth.config.properties.KafkaTopicProperties
import org.apache.kafka.clients.admin.NewTopic
import org.springframework.context.annotation.Bean
import org.springframework.kafka.config.TopicBuilder

/**
 * Configuration of kafka topics
 *
 * @author diagorn
 */
class KafkaTopicConfig(
    private val kafkaTopicProperties: KafkaTopicProperties
) {

    @Bean
    fun newUserTopic(): NewTopic = TopicBuilder
        .name(kafkaTopicProperties.newUser)
        .partitions(kafkaTopicProperties.newUserPartitionNum)
        .build()

    @Bean
    fun editUserTopic(): NewTopic = TopicBuilder
        .name(kafkaTopicProperties.editUser)
        .partitions(kafkaTopicProperties.editUserPartitionNum)
        .build()
}